api/buckets/service.go

```diff
@@ -917,15 +1104,35 @@ func (s *Service) PushPath(server pb.APIService_PushPathServer) error {
 	if err != nil {
 		return err
 	}
+
 	buck := &tdb.Bucket{}
-	err = s.Buckets.Get(server.Context(), dbID, key, buck, tdb.WithToken(dbToken))
+	err = s.Buckets.GetSafe(server.Context(), dbID, buckKey, buck, tdb.WithToken(dbToken))
 	if err != nil {
 		return err
 	}
 	if root != "" && root != buck.Path {
 		return status.Error(codes.FailedPrecondition, buckets.ErrNonFastForward.Error())
 	}
 
+	buck.UpdatedAt = time.Now().UnixNano()
+	buck.SetMetadataAtPath(filePath, tdb.Metadata{
+		UpdatedAt: buck.UpdatedAt,
+	})
+	buck.UnsetMetadataWithPrefix(filePath + "/")
```

Comment:
In the case where this push is overwriting a directory, we want to wipe out the metadata for those killed paths.